sound/soc/sof/loader.c

```diff
@@ -20,6 +20,10 @@ static int get_ext_windows(struct snd_sof_dev *sdev,
 {
 	const struct sof_ipc_window *w =
 		container_of(ext_hdr, struct sof_ipc_window, ext_hdr);
+	/*
+	 * should by struct_size instead of sizeof to keep backward 
+	 * compatibility with ABI 3.16.0 and below
+	 */
 	size_t w_size = struct_size(w, window, w->num_windows);
```

Comment:
mmh, correct me if I'm wrong. This calculation now makes `w_size = sizeof(struct sof_ipc_window) + sizeof(w->window) * w->num_windows` so now it includes `SOF_IPC_MAX_ELEMS` of `struct sof_ipc_window_elem` plus `w->num_windows` of them? Is this really what we need?

Reply:
Yes, I should use `ext_hdr->hdr.size` to get proper size.